src/borrow_tracker/tree_borrows/tree/tests.rs

```diff
@@ -75,151 +75,109 @@ fn protected_enforces_noalias() {
-                (AccessRelatedness::StrictChildAccess, AccessRelatedness::DistantAccess),
-            AccessRel2MutuallyForeign::ChildY =>
-                (AccessRelatedness::DistantAccess, AccessRelatedness::StrictChildAccess),
-            AccessRel2MutuallyForeign::Other =>
-                (AccessRelatedness::DistantAccess, AccessRelatedness::DistantAccess),
+            RelPosXY::MutuallyForeign => true,
+            RelPosXY::XChildY => false,
         }
     }
 }
 
-impl Into2Rel for AccessRel2XChildY {
-    const LEGEND: &'static str = "x is a strict child of y";
-    const THROUGH_X: Self = Self::ChildBoth;
-    const THROUGH_Y: Self = Self::ChildY;
-    // Parent `y` has to already exist upon creation of `x` its child.
-    const ALLOWS_X_RETAG_FIRST: bool = false;
-    fn into_pair(self) -> (AccessRelatedness, AccessRelatedness) {
-        match self {
-            AccessRel2XChildY::ChildBoth =>
-                (AccessRelatedness::StrictChildAccess, AccessRelatedness::StrictChildAccess),
-            AccessRel2XChildY::ChildY =>
-                (AccessRelatedness::AncestorAccess, AccessRelatedness::StrictChildAccess),
-            AccessRel2XChildY::ChildNeither =>
-                (AccessRelatedness::DistantAccess, AccessRelatedness::DistantAccess),
+impl PtrSelector {
+    fn rel_pair(self, xy_rel: RelPosXY) -> (AccessRelatedness, AccessRelatedness) {
+        match xy_rel {
+            RelPosXY::MutuallyForeign =>
+                match self {
+                    PtrSelector::X =>
+                        (AccessRelatedness::StrictChildAccess, AccessRelatedness::DistantAccess),
+                    PtrSelector::Y =>
+                        (AccessRelatedness::DistantAccess, AccessRelatedness::StrictChildAccess),
+                    PtrSelector::Other =>
+                        (AccessRelatedness::DistantAccess, AccessRelatedness::DistantAccess),
+                },
+            RelPosXY::XChildY =>
+                match self {
+                    PtrSelector::X =>
+                        (AccessRelatedness::StrictChildAccess, AccessRelatedness::StrictChildAccess),
+                    PtrSelector::Y =>
+                        (AccessRelatedness::AncestorAccess, AccessRelatedness::StrictChildAccess),
+                    PtrSelector::Other =>
+                        (AccessRelatedness::DistantAccess, AccessRelatedness::DistantAccess),
+                },
         }
     }
 }
 
-impl<Rel2: Into2Rel> fmt::Display for Access2<Rel2> {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let kind_text = match self.kind {
-            AccessKind::Read => "read",
-            AccessKind::Write => "write",
-        };
-        write!(f, "{kind_text} {}", self.rel2)
-    }
-}
-
 /// Arbitrary access parametrized by the relative position of `x` and `y`
 /// to each other.
 #[derive(Debug, Clone, Copy)]
-struct Access2<Rel2> {
-    rel2: Rel2,
+struct Access2 {
```

Comment:
I think we should get rid of all of these "2" suffices, they are confusing. What about `TestAccess` and `TestEvent`?